### ISSUE TITLE
E1101: (Flask) Method 'jinja_env' has no 'add_extension', 'filter' member (no-member)

### ISSUE THREAD

**Issue by bimalkant-lauhny** [PyCQA/pylint#2563]
### Steps to reproduce
1. Run `pylint hello.py` (after installing flask 1.0.2), where `hello.py` is a simple flask app as below:
```python
from flask import Flask
app = Flask(__name__)
app.jinja_env.add_extension('jinja2.ext.do')
app.jinja_env.filters['somefilter'] = lambda s: s

@app.route('/')
def hello_world():
    return 'Hello, World!'
```

### Current behavior
pylint outputs the following for the file above:
```shell
************* Module hello
hello.py:3:0: E1101: Method 'jinja_env' has no 'add_extension' member (no-member)
hello.py:4:0: E1101: Method 'jinja_env' has no 'filters' member (no-member)
```

### Expected behavior
It should give no error because `add_extension` and `filters` are members of `Flask.jinja_env`.

### pylint --version output
```shell
pylint 2.1.1
astroid 2.0.4
python 3.6.6
```
### Check the docs below for `jinja environment`
[Flask.jinja_env](http://flask.pocoo.org/docs/1.0/api/?highlight=jinja_env#flask.Flask.jinja_env)
[Flask.jinja_env.filters](http://flask.pocoo.org/docs/1.0/templating/#registering-filters)
[jinja2.Environment.add_extension](http://jinja.pocoo.org/docs/2.10/api/#jinja2.Environment.add_extension)
* Also, `jinja_env` is a cached property. Somewhat like [this](https://github.com/pydanny/cached-property).

### workaround
apart from disabling, if we change code as follows, pylint shows no errors.
```python
app.jinja_env().add_extension('jinja2.ext.do')
app.jinja_env().filters['somefilter'] = lambda s: s
```

**Comment by aryanptl01** [comment]
Why does the workaround work?

**Comment by gdubicki** [comment]
The workaround did not for me. I got this error then:
```
E1102: app.jinja_env is not callable (not-callable)
```
I used this `pylintrc` to disable this check:
```
[tool.pylint."messages control"]

disable = no-member
```

**Comment by Pierre-Sassoulas** [comment]
The solution until we make astroid better at understanding flask code is to use the ``generated-members`` option see https://pylint.pycqa.org/en/latest/user_guide/messages/error/no-member.html.
